google/cloud/aiplatform/tensorboard/uploader.py

```diff
@@ -335,6 +337,23 @@ def _create_additional_senders(self) -> Dict[str, RequestSender]:
             Mapping from plugin name to Sender.
         """
         additional_senders = {}
+        if "profile" in self._allowed_plugins:
+            if not self._one_shot:
+                logger.warning("Profile plugin currently only supported for one shot.")
```

Comment:
I think it's better to do this validation as early as possible, most likely in the constructor of TensorBoardUploader.

Could you also specify this in the doc string of the "allowed_plugins" argument?

I am also inclined to throw an exception instead of log-warn.